test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/BuildServerIntegrationTest.cs

```diff
@@ -143,28 +144,55 @@ public async Task Build_ServerConnectionMutexCreationFails_FallsBackToInProcessR
             // it reaches server creation part.
         }
 
-        // Skipping on linux/mac because of https://github.com/aspnet/Razor/issues/2507.
+        // Skipping on MacOS because of https://github.com/dotnet/corefx/issues/33141.
         [ConditionalFact]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
         [InitializeTestProject("SimpleMvc")]
         public async Task ManualServerShutdown_NoPipeName_ShutsDownServer()
         {
-            var toolAssembly = typeof(Application).Assembly.Location;
-            var result = await DotnetMSBuild(
-                "Build",
-                $"/p:_RazorForceBuildServer=true /p:_RazorToolAssembly={toolAssembly}",
-                suppressBuildServer: true); // We don't want to specify a pipe name
+            // Publish rzc tool to a temporary path
+            var solutionRoot = TestPathUtilities.GetSolutionRootDirectory("Razor");
+            var toolAssemblyDirectory = Path.Combine(solutionRoot, "src", "Microsoft.AspNetCore.Razor.Tools");
+            var toolAssemblyPath = Path.Combine(toolAssemblyDirectory, "Microsoft.AspNetCore.Razor.Tools.csproj");
+            var projectDirectory = new TestProjectDirectory(solutionRoot, toolAssemblyDirectory, toolAssemblyPath);
+            var publishDir = Path.Combine(Path.GetTempPath(), "Razor", Path.GetRandomFileName(), "RzcPublish");
+            var publishResult = await MSBuildProcessManager.RunProcessAsync(projectDirectory, $"/t:Publish /p:PublishDir=\"{publishDir}\"");
 
-            Assert.BuildPassed(result);
+            try
+            {
+                // Run the build using the published tool
+                var toolAssembly = Path.Combine(publishDir, "rzc.dll");
+                var result = await DotnetMSBuild(
+                    "Build",
+                    $"/p:_RazorForceBuildServer=true /p:_RazorToolAssembly={toolAssembly}",
+                    suppressBuildServer: true); // We don't want to specify a pipe name
 
-            // Shutdown the server
-            var output = new StringWriter();
-            var error = new StringWriter();
-            var application = new Application(CancellationToken.None, Mock.Of<ExtensionAssemblyLoader>(), Mock.Of<ExtensionDependencyChecker>(), (path, properties) => Mock.Of<PortableExecutableReference>(), output, error);
-            var exitCode = application.Execute("shutdown", "-w");
+                Assert.BuildPassed(result);
 
-            Assert.Equal(0, exitCode);
-            Assert.Contains("shut down completed", output.ToString());
+                // Manually shutdown the server
+                var processStartInfo = new ProcessStartInfo()
+                {
+                    WorkingDirectory = publishDir,
+                    UseShellExecute = false,
+                    RedirectStandardError = true,
```

Comment:
Do you need the redirects since you're no longer capturing output?

Reply:
Not really but the `MsBuildProcessManager` is wired up to capture this output and expects this to be set to true.